systemtest/src/main/java/io/strimzi/systemtest/performance/report/PerformanceReporter.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.performance.report;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.strimzi.api.kafka.model.kafka.Kafka;
+import io.strimzi.systemtest.performance.PerformanceConstants;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.storage.TestStorage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Reports performance metrics from tests related to Strimzi Kafka operators.
+ * <p>
+ * This class provides functionality to log performance data gathered during performance tests,
+ * including operation durations, Kafka configuration, and custom performance metrics defined
+ * in {@link PerformanceConstants}. The performance data is logged to files organized by test
+ * use case and configuration parameters such as max batch size and max linger time.
+ * <p>
+ * Key Features:
+ * <ul>
+ *     <li>Logs detailed performance metrics and Kafka configurations in a structured format.</li>
+ *     <li>Supports logging of complex data types, including long arrays and maps, with special
+ *     handling for serialization.</li>
+ *     <li>Organizes logs by test use case and configuration parameters, facilitating easier
+ *     analysis of performance under different conditions.</li>
+ * </ul>
+ */
+public class PerformanceReporter {
+    private static final Logger LOGGER = LogManager.getLogger(PerformanceReporter.class);
+
+    public static void logPerformanceData(
```

Comment:
javadoc here would make the method usecase clearer (same applies for all other methods)